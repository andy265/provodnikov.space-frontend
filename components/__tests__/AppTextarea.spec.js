const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
const AppTextarea = require('~/components/AppTextarea')

describe('AppTextarea.vue', () => {
  it('has same HTML structure', () => {
    let wrapper = shallow(AppTextarea, {
      propsData: {
        value: 'some text'
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('emit input event on change', () => {
    let wrapper = shallow(AppTextarea, {
      propsData: {
        value: 'some text'
      }
    })
    wrapper.find('textarea').element.value = 'new text'
    wrapper.find('textarea').trigger('input')
    expect(wrapper.emitted().input[0][0]).toBe('new text')
  })
})
