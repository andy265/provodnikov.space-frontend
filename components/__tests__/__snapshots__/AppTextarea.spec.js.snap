// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTextarea.vue has same HTML structure 1`] = `<textarea value="some text" class="app-textarea">some text</textarea>`;
